crates/starfish/core/src/transactions_synchronizer.rs

```diff
@@ -928,12 +955,16 @@ impl<C: NetworkClient, V: BlockVerifier, D: CoreThreadDispatcher>
 
         tokio::pin!(fetcher_timeout);
 
-        // Track the number of concurrent requests
+        // Increase the number of concurrent requests
+        let number_of_requests: i64 = request_futures
+            .len()
+            .try_into()
+            .expect("We should expect a correct conversion from request_futures.len() to i64");
         context
             .metrics
             .node_metrics
             .transactions_synchronizer_inflight_requests
-            .set(request_futures.len() as i64);
+            .add(number_of_requests);
```

Comment:
This should be `set` to reflect the number of inflight requests at a given point in time. Also, looks like we should make sure that this metric is tracked correctly overall in other places as it looks a bit suspicious now that I look at it.